starlite/app.py

```diff
@@ -512,26 +600,52 @@ def _add_node_to_route_map(self, route: BaseRoute) -> RouteMapNode:
         self._configure_route_map_node(route, current_node)
         return current_node
 
-    def _add_route_to_handler_index(self, route: BaseRoute) -> None:
-        """Maps route handler names to urls.
+    def _get_route_handlers(self, route: BaseRoute) -> List["RouteHandlerType"]:
+        """Retrieve handler(s) as a list for given route."""
+        route_handlers: List["RouteHandlerType"] = []
+        if isinstance(route, (WebSocketRoute, ASGIRoute)):
+            route_handlers.append(route.route_handler)
+        else:
+            route_handlers.extend(cast("HTTPRoute", route).route_handlers)
+
+        return route_handlers
+
+    def _get_handler_qualname(self, handler: "RouteHandlerType") -> str:
+        """Retrieves __qualname__ for passed route handler."""
+        if handler.fn and hasattr(handler.fn, "__qualname__"):
```

Comment:
this is a place to use `cast` because there is no way to get a handler without `.fn` - it will raise an exception before reaching this point. 

```python
   return cast("AnyCallable", handler.fn).__qualname__
```

Or something along these lines.

Reply:
@Goldziher I wasn't sure if `handler.fn` is not None at all cases during route tree building but since you say so we can drop `not handler.fn` check.

However, we can not get rid of `hasattr(handler.fn, "__qualname__")` since `.fn` is a callable and an instance of a class with `__call__` method is a callable too but it has no `__qualname__` attr.

I think If I remove this check the method would fail for `StaticFiles` which is a Starlette class with `__call__` method. I haven't checked it manually though but I explored the code in order to see how I can create a template function which generates links to static files.